src/Tizen.NUI.Scene3D/src/public/ModelMotion/MotionData.cs

```diff
@@ -16,109 +16,384 @@
     /// </remark>
     [EditorBrowsable(EditorBrowsableState.Never)]
-    public class MotionData : IDisposable
+    public class MotionData : BaseHandle
     {
-        private List<(MotionIndex, MotionValue)> motionValues { get; set; } = null;
+        private EventHandler loadCompletedEventHandler;
+        private LoadCompletedCallbackType loadCompletedCallback;
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate void LoadCompletedCallbackType(IntPtr motionData);
 
         /// <summary>
-        /// Owned motion value list.
+        /// Create an initialized motion data.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public MotionData() : this(Interop.MotionData.MotionDataNew(), true)
+        {
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+        }
+
+        /// <summary>
+        /// Create an initialized motion data with duration.
+        /// </summary>
+        /// <param name="durationSeconds">Duration of this motion data when it be generated as Animation.</param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public MotionData(float durationSeconds) : this(Interop.MotionData.MotionDataNew(durationSeconds), true)
+        {
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+        }
+
+        /// <summary>
+        /// Copy constructor.
+        /// </summary>
+        /// <param name="motionData">Source object to copy.</param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public MotionData(MotionData motionData) : this(Interop.MotionData.NewMotionData(MotionData.getCPtr(motionData)), true)
+        {
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+        }
+
+        /// <summary>
+        /// Assignment operator.
+        /// </summary>
+        /// <param name="motionData">Source object to be assigned.</param>
+        /// <returns>Reference to this.</returns>
+        internal MotionData Assign(MotionData motionData)
+        {
+            MotionData ret = new MotionData(Interop.MotionData.MotionDataAssign(SwigCPtr, MotionData.getCPtr(motionData)), false);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            return ret;
+        }
+
+        internal MotionData(global::System.IntPtr cPtr, bool cMemoryOwn) : base(cPtr, cMemoryOwn)
+        {
+        }
+
+        /// <summary>
+        /// Get or set the duration of this motion data as seconds.
         /// </summary>
         [EditorBrowsable(EditorBrowsableState.Never)]
-        [Obsolete("Do not use this MotionValues property directly. Use Add and GetIndex, GetValue instead.")]
-        public List<(MotionIndex, MotionValue)> MotionValues
+        public float Duration
```

Comment:
How do you think which is better Duration or DurationMillisecond?
NUI usually uses later naming in animation.

Reply:
The name of property use still `Duration`. I'll keep current logic.